### DISCUSSION TITLE
Resource pack item model transform

### DISCUSSION THREAD

**Discussion by PixlRainbow** [MineLittlePony/MineLittlePony#292] [Q&A] [answered]
# Background 
I have made a custom model for the spyglass in BlockBench and adjusted transform configuration (orientation and positioning) for how it displays in first person, third person and on head, based on the previews for human models shown inside BlockBench.
Testing the custom model with a human model in-game seems to yield the expected orientations and positions when the spyglass is held in hand or used.
Testing with a pony model in-game, the third person display seems to work fine, **but in first person the custom spyglass model is being held backwards, and when used on the head, it is sideways.**
This doesn't happen with the vanilla spyglass model.
# Question
Are there any tips for how to configure the model in a way that it would be cross compatible between human and pony models?
Do I just have to make a dedicated version of the resource pack for mine little pony?

**Comment by Sollace** [comment, marked as answer]
You may have to make a dedicated version unfortunately. And even then it is possible to have human and pony models appearing at the same time with the mod so that may not address the issue fully.

Typically, the transforms mine little pony applies assume the vanilla model transforms (not to mention the base game does some rotations on its own outside of what you can control through the model) and it tries to keep them as close to the original when you're not playing as a unicorn with first/third-person magic turned on.

**Reply by PixlRainbow** [reply]
Noted. But I did observe something interesting: if you change the resource pack while in-world (through the pause menu), mine little pony doesn't apply its own transforms until the item model leaves the view and comes back (either by switching held item, or switching perspective)
